### PR TITLE
Test all PCA param combinations

### PR DESCRIPTION
**Pull request by flying-sheep**
This replaces the random choice with iterating over all combinations. That way, if you want to debug a certain combination, you can just do so instead of rerunning the test and hoping it gets picked.

Sadly AFAIK it’s not possible to have a fixture that depends on other fixture values and generates a variable amount of values depending on their arguments: either you have `fixture(params=some_list)` which creates `len(some_list)` values or not, then it creates one. Therefore I had to get rid of the fixtures and use a static list instead.

It’s not that much slower to run them all:

before: 24 passed, 6 xfailed in 1.81s 
after: 56 passed, 14 xfailed in 3.41s

### CODE REVIEW

File: tests/test_pca.py [hunk 1]
```diff
@@ -83,75 +81,105 @@ def wrapper(a: np.ndarray) -> DaskArray:
...
 
 
-@pytest.fixture(
-    params=[
-        param_with(at, marks=[needs.dask_ml]) if "dask" in at.id else at
-        for at in ARRAY_TYPES_SPARSE_DASK_UNSUPPORTED
-    ]
-)
-def array_type(request: pytest.FixtureRequest):
+def conv_at(array_type):
```
Comment (ilan-gold) [inline tests/test_pca.py]: type please but also name of the function to `maybe_convert_array_to_dask` or something?

File: tests/test_pca.py [hunk 2]
```diff
@@ -83,75 +81,105 @@ def wrapper(a: np.ndarray) -> DaskArray:
...
-        and zero_center
-        and svd_solver == "lobpcg"
-    ):
-        warn_pat_expected = r"legacy code"
+SVDSolver = Literal["auto", "full", "arpack", "randomized", "tsqr", "lobpcg"]
+
 
-    return (svd_solver, warn_pat_expected)
+def gen_pca_params(
+    *, array_type, svd_solver_type: Literal[None, "valid", "invalid"], zero_center: bool
```
Comment (ilan-gold) [inline tests/test_pca.py]: Ok, so there's an issue with typing `array_type`?

File: tests/test_pca.py [hunk 3]
```diff
@@ -324,7 +352,7 @@ def test_pca_n_pcs():
 
 # We use all ARRAY_TYPES here since this error should be raised before
```
Comment (ilan-gold) [inline tests/test_pca.py]:
```suggestion
# We use all ARRAY_TYPES_ALL_SUPPORTED here since this error should be raised before
```

**Review by ilan-gold — APPROVED** [review]

File: tests/test_pca.py (hunk 2 above)
Comment (flying-sheep) [reply tests/test_pca.py]: no, I can do it!

**Pull request merged**